packages/components/src/internal/components/search/utils.ts

```diff
@@ -309,6 +343,13 @@ export function getUpdatedChooseValuesFilter(
         newCheckedValues.push(v === EMPTY_VALUE_DISPLAY ? '' : v);
     });
 
+    // skip optimization for ANCESTOR_MATCHES_ALL filter type
+    if (isAncestorMatchesAllFilter) {
+        if ((newValue === ALL_VALUE_DISPLAY && !check) || newCheckedValues.length === 0)
+            return Filter.create(fieldKey, [], ANCESTOR_MATCHES_ALL_OF_FILTER_TYPE);
```

Comment:
I'm probably not reading this properly, but doesn't this result in a "no filter value" error for the filter?  Seems like we would return `null` here but maybe we want to show the error.

Reply:
The idea is to retain the filter type when user is on Choose Value tab . If we return null, the filter will get automatically get changed to equals (one of) as user starts to select value.